maryprazyan/candlestick
# Make the system scaleable

## What was done
- Integrated `Hazelcast` distributed cache to handle `openCandlesticks`. It should handle scalability issues 
- Integrated RabbitMQ for distributed queuing mechanism. Now the `WebSocketManager` publishes the queues via `RabbitMQ` and process methods are listening and handling them. 
- Removed unused files
- Made the server port configurable so the user can run multiple instances of the application at the same time

With the usage of these frameworks, the system is capable of handling more data and making it more distributes

## Testing
- The Data processor tests have been updated due to `Hazelcast` integration
- To test the app
  - run docker-compose up
  - ./gradlew run (or can run multiple instances: PORT=YOUR_PORT_NUMBER ./gradlew run)
  - check the logs to see that the instruments and quotes are streamed
  - open http://localhost:9000/swagger-ui/index.html in a web browser.
  - use the /instruments endpoint to get a list of instruments, noting down a valid ISIN.
  - wait for at least one minute so that the candlesticks can be generated for your chosen instrument (or monitor the logs to check if an instrument is deleted which had at least one quote).
  - use the /candlesticks endpoint with your chosen ISIN to retrieve candlestick data.
